COMPLETE_DOCUMENTATION.md

```diff
@@ -0,0 +1,3471 @@
+
+You can enable client statistics and set a non-default period in seconds as follows:
+
+**Declarative Configuration:**
+
+```json
+{
+    "properties": {
+        "hazelcast.client.statistics.enabled": true,
+        "hazelcast.client.statistics.period.seconds": 4
+    }
+}
+```
+
+**Programmatic Configuration:**
+
+```javascript
+var config = new Config.ClientConfig();
+config.properties['hazelcast.client.statistics.enabled'] = true;
+config.properties['hazelcast.client.statistics.period.seconds'] = 4;
+```
+
+After enabling the client statistics, you can monitor your clients using Hazelcast Management Center. Please refer to the [Monitoring Clients section](https://docs.hazelcast.org/docs/management-center/latest/manual/html/index.html#monitoring-clients) in the Hazelcast Management Center Reference Manual for more information on the client statistics.
+
+### 7.9.2. Logging Configuration
+
+ By default, Hazelcast Node.js client uses a default logger which logs to the `stdout` with the `INFO` log level. You can change the log level using the `'hazelcast.logging.level'` property of the `ClientConfig.properties`.
+ 
+Below is an example of the logging configuration with the `OFF` log level which disables logging.
+ 
+```javascript
+cfg.properties['hazelcast.logging.level'] = LogLevel.OFF;
+```
+ 
+ You can also implement a custom logger depending on your needs. Your custom logger must have `log`, `error`, `warn`, `info`, `debug`, `trace` methods. After implementing it, you can use your custom logger using the `customLogger` property of `ClientConfig`
+  
+See the following for a custom logger example.
+ 
+```javascript
+var winstonAdapter = {
+    logger: new (winston.Logger)({
+        transports: [
+            new (winston.transports.Console)()
+        ]
+    }),
+
+    levels: [
+        'error',
+        'warn',
+        'info',
+        'debug',
+        'silly'
+    ],
+
+    log: function (level, objectName, message, furtherInfo) {
+        this.logger.log(this.levels[level], objectName + ': ' + message, furtherInfo);
+    },
+
+    error: function (objectName, message, furtherInfo) {
+        this.log(LogLevel.ERROR, objectName, message, furtherInfo);
+    },
+
+    warn: function (objectName, message, furtherInfo) {
+        this.log(LogLevel.WARN, objectName, message, furtherInfo);
+    },
+
+    info: function (objectName, message, furtherInfo) {
+        this.log(LogLevel.INFO, objectName, message, furtherInfo);
+    },
+
+    debug: function (objectName, message, furtherInfo) {
+        this.log(LogLevel.DEBUG, objectName, message, furtherInfo);
+    },
+
+    trace: function (objectName, message, furtherInfo) {
+        this.log(LogLevel.TRACE, objectName, message, furtherInfo);
+    }
+
+};
+cfg.customLogger = winstonAdapter;
+```
+
+Note that it is not possible to configure custom logging via declarative configuration.
+
+# 8. Development and Testing
```

Comment:
The development section in the DOCUMENTATION.md talks about linking the client package so that it can be used in other projects. Do you think that we should put these changes into README before removing them ?

Reply:
I think we can remove `npm link` part completely, as it's not that important.